### PR TITLE
Make bdbje transaction lock configurable

### PR DESCRIPTION
**Pull request by xdev-developer**
Signed-off-by: Den Kovalevsky <xdev.developer@gmail.com>

### CODE REVIEW

**Comment by mbrukman** [comment]
@xdev-developer – thanks for the patch! Before we can accept it, please follow the instructions in [`CONTRIBUTING.md`](https://github.com/JanusGraph/janusgraph/blob/master/CONTRIBUTING.md) to sign and submit a CCLA or ICLA, as appropriate.

**Comment by xdev-developer** [comment]
done. CLA submitted.

File: janusgraph-berkeleyje/src/main/java/org/janusgraph/diskstorage/berkeleyje/BerkeleyJEStoreManager.java [hunk 1]
```diff
@@ -105,12 +112,13 @@ public BerkeleyJEStoreManager(Configuration configuration) throws BackendExcepti
...
     }
 
-    private void initialize(int cachePercent) throws BackendException {
+    private void initialize(int cachePercent, long txLockTimeout) throws BackendException {
         try {
             EnvironmentConfig envConfig = new EnvironmentConfig();
             envConfig.setAllowCreate(true);
             envConfig.setTransactional(transactional);
             envConfig.setCachePercent(cachePercent);
+            envConfig.setLockTimeout(txLockTimeout, TimeUnit.MILLISECONDS);
```
Comment (amcp) [inline janusgraph-berkeleyje/src/main/java/org/janusgraph/diskstorage/berkeleyje/BerkeleyJEStoreManager.java]: please add a test that demonstrates the new lock timeout config

Comment (xdev-developer) [inline janusgraph-berkeleyje/src/main/java/org/janusgraph/diskstorage/berkeleyje/BerkeleyJEStoreManager.java]: ok.

**Comment by jerryjch** [comment]
LGTM. 
Eventually, to broadly support more native properties for a particular backend, we want to be able to just pass-thru the native properties.  For example, storage.berkeleyje.xxx in the properties file will be passed thru to berkeleyje.

File: janusgraph-berkeleyje/src/main/java/org/janusgraph/diskstorage/berkeleyje/BerkeleyJEStoreManager.java (hunk 1 above)
Comment (pluradj) [inline janusgraph-berkeleyje/src/main/java/org/janusgraph/diskstorage/berkeleyje/BerkeleyJEStoreManager.java]: @xdev-developer any updates on this?

File: janusgraph-berkeleyje/src/main/java/org/janusgraph/diskstorage/berkeleyje/BerkeleyJEStoreManager.java [hunk 2]
```diff
@@ -69,6 +70,11 @@
             ConfigOption.Type.MASKABLE,  String.class,
             IsolationLevel.REPEATABLE_READ.toString(), disallowEmpty(String.class));
 
+    public static final ConfigOption<Long> LOCK_TIMEOUT =
+        new ConfigOption<Long>(BERKELEY_NS,"lock-timeout",
```
Comment (analytically) [inline janusgraph-berkeleyje/src/main/java/org/janusgraph/diskstorage/berkeleyje/BerkeleyJEStoreManager.java]: spacing

File: janusgraph-berkeleyje/src/main/java/org/janusgraph/diskstorage/berkeleyje/BerkeleyJEStoreManager.java [hunk 3]
```diff
@@ -69,6 +70,11 @@
...
+            "Lock timeout for all transactional and non-transactional operations",
```
Comment (pluradj) [inline janusgraph-berkeleyje/src/main/java/org/janusgraph/diskstorage/berkeleyje/BerkeleyJEStoreManager.java]: Description should mention that the units for the value is milliseconds, and that the default is 500ms and maximum is 75min according to the BerkeleyJE [LOCK_TIMEOUT docs](https://docs.oracle.com/cd/E17277_02/html/java/com/sleepycat/je/EnvironmentConfig.html#LOCK_TIMEOUT).

**Comment by farodin91** [comment]
@xdev-developer Do you want to do anything fix all requested changes? Otherwise, I would close this PR.

**Pull request closed without merging**